### PR TITLE
Fix: update test to validate pmapEvent in successMessenger

### PR DESCRIPTION
**Pull request by sailorlqh**
fix: #97

### CODE REVIEW

**Review by sethvargo — APPROVED** [review]

File: pkg/server/event_handler_test.go [hunk 1]
```diff
@@ -468,12 +498,42 @@ func (p *successProcessor) Process(_ context.Context, m *structpb.Struct) error
 }
 
 // failMessenger Send always fail.
-type failMessenger struct{}
 
-func (m *failMessenger) Send(_ context.Context, _ *v1alpha1.PmapEvent) error {
+type testMessanger interface {
+	Messenger
+	getPmapEvent() *v1alpha1.PmapEvent
+}
```
Comment (yolocs) [inline pkg/server/event_handler_test.go:505]: We don't need this interface as long as our fake messenger implements the `Messenger` interface.

File: pkg/server/event_handler_test.go [hunk 2]
```diff
@@ -468,12 +498,42 @@ func (p *successProcessor) Process(_ context.Context, m *structpb.Struct) error
...
+
+type failMessenger struct {
+	gotPmapEvent *v1alpha1.PmapEvent
+}
```
Comment (yolocs) [inline pkg/server/event_handler_test.go:509]: Can we just have one fake messenger? The only difference between the two is one requires error. You achieve the same with a field. E.g.

```go
type testMessenger struct {
  returnErr error
  gotEvent *v1alpha1.PmapEvent
}

func (m *testMessenger) Send(_ context.Context, p *v1alpha1.PmapEvent) error {
  m.gotEvent = p
  return m.returnErr
}

// We should get rid of these in a separate refactor
// See https://github.com/abcxyz/pmap/issues/96
func (m *testMessenger) Cleanup() error {
	return nil
}
```

**Review by yolocs — CHANGES_REQUESTED** [review]

**Review by yolocs — APPROVED** [review]

File: pkg/server/event_handler_test.go [hunk 3]
```diff
@@ -388,6 +427,18 @@ isOK: true`),
...
 				t.Errorf("Process(%+v) got unexpected error substring: %v", tc.name, diff)
 			}
+
+			cmpOpts := []cmp.Option{
+				protocmp.Transform(),
+				protocmp.IgnoreFields(&v1alpha1.PmapEvent{}, "payload"),
+				cmpopts.IgnoreUnexported(v1alpha1.GitHubSource{}),
+				cmpopts.IgnoreUnexported(timestamppb.Timestamp{}),
+			}
+			if tc.wantPmapEvent != nil {
```
Comment (yolocs) [inline pkg/server/event_handler_test.go:437]: I think you don't need this `if`. cmp.Diff can handle nil

File: pkg/server/event_handler_test.go [hunk 4]
```diff
@@ -388,6 +446,16 @@ isOK: true`),
 			if diff := testutil.DiffErrString(gotErr, tc.wantErrSubstr); diff != "" {
 				t.Errorf("Process(%+v) got unexpected error substring: %v", tc.name, diff)
 			}
+
+			cmpOpts := []cmp.Option{
+				protocmp.Transform(),
+				protocmp.IgnoreFields(&v1alpha1.PmapEvent{}, "payload"),
+				cmpopts.IgnoreUnexported(v1alpha1.GitHubSource{}),
```
Comment (capri-xiyue) [inline pkg/server/event_handler_test.go:453]: Why do we need these two  IgnoreUnexported?

Comment (sailorlqh) [reply pkg/server/event_handler_test.go]: Forgot to remove it. Just removed.

**Review by capri-xiyue — APPROVED** [review]

**Pull request merged**